wrangler-service/src/main/java/co/cask/wrangler/service/connections/ConnectionService.java

```diff
@@ -234,12 +237,33 @@ public void list(HttpServiceRequest request, HttpServiceResponder responder,
         }
         return false;
       });
-      responder.sendJson(new ConnectionResponse<>(connections, connectionTypeConfig.getDefaultConnection()));
+      Optional<Connection> defaultConnection = getDefaultConnection(connections, connectionTypeConfig);
+      responder.sendJson(
+        new ConnectionResponse<>(connections,
+                                 defaultConnection.isPresent() ? defaultConnection.get().getName() : null));
     } catch (Exception e) {
       error(responder, e.getMessage());
     }
   }
 
+  private boolean isDefaultConnection(String connectionId) {
```

Comment:
the use of 'default' is confusing because there is an actual single default connection. A better name could be 'canBeDeleted', as that's really what it's being used for.

I also think it's more clear if you just put the logic in the delete method instead of having a separate method, since it's basically a single line and is only called from one place.